st2common/st2common/transport/bootstrap_utils.py

```diff
@@ -56,6 +57,9 @@ def _do_register_exchange(exchange, connection, channel, retry_wrapper):
         LOG.exception('Failed to register exchange : %s.', exchange.name)
 
 
+@retrying.retry(
```

Comment:
Is it possible to only catch more specific exceptions aka connection errors?

Otherwise it could potentially mask some other / real issues...